NetworkingTests/Webservice/FormEncoderTests.swift

```diff
@@ -0,0 +1,283 @@
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleInt16Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(Int16(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleInt8Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(Int8(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleInt32Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(Int32(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleInt64Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(Int64(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleUInt32Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(UInt32(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testSingleUInt64Encoding() {
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(UInt64(1))
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
+        } catch {
+            XCTFail("wrong error")
+        }
+    }
+
+    func testUnkeyeyedArrayCustomTypeEncoding() {
+        let iPhone = Product(name: "iPhone X", price: 1_000, info: "Our best iPhone yet!")
+        let macBook = Product(name: "Mac Book Pro", price: 2_000, info: "Early 2019")
+        let watch = Product(name: "Apple Watch", price: 500, info: "Series 4")
+
+        let products =  [ iPhone, macBook, watch]
+
+        let formEncoder = FormEncoder()
+        do {
+            let formString = try formEncoder.stringEncode(products)
+            print(formString)
+        } catch FormEncoder.Error.unsupportedDataFormat {
+            print("test passed")
```

Comment:
There is an XCTAssertThrows, should be using that if validating errors are thrown